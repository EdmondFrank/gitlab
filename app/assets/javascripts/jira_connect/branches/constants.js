import { __ } from '~/locale';

export const BRANCHES_PER_PAGE = 20;
export const PROJECTS_PER_PAGE = 20;

export const I18N_NEW_BRANCH_FORM = {
  pageTitle: __('New branch'),
  labels: {
    projectDropdown: __('Project'),
    branchNameInput: __('Branch name'),
    sourceBranchDropdown: __('Source branch'),
  },
  formSubmitButtonText: __('Create branch'),
};

export const CREATE_BRANCH_ERROR_GENERIC = __('Failed to create branch. Please try again.');
export const CREATE_BRANCH_ERROR_WITH_CONTEXT = __('Failed to create branch.');

export const CREATE_BRANCH_SUCCESS_ALERT = {
  title: __('New branch was successfully created.'),
  message: __('You can now close this window and return to Jira.'),
};
