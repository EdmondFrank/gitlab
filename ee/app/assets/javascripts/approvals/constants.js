import { __, s__ } from '~/locale';

export const TYPE_USER = 'user';
export const TYPE_GROUP = 'group';
export const TYPE_HIDDEN_GROUPS = 'hidden_groups';

export const BRANCH_FETCH_DELAY = 250;
export const ANY_BRANCH = {
  id: null,
  name: __('Any branch'),
};

export const RULE_TYPE_FALLBACK = 'fallback';
export const RULE_TYPE_REGULAR = 'regular';
export const RULE_TYPE_REPORT_APPROVER = 'report_approver';
export const RULE_TYPE_CODE_OWNER = 'code_owner';
export const RULE_TYPE_ANY_APPROVER = 'any_approver';
export const RULE_NAME_ANY_APPROVER = 'All Members';

export const VULNERABILITY_CHECK_NAME = 'Vulnerability-Check';
export const LICENSE_CHECK_NAME = 'License-Check';
export const COVERAGE_CHECK_NAME = 'Coverage-Check';

export const LICENSE_SCANNING = 'license_scanning';

export const APPROVAL_RULE_CONFIGS = {
  [VULNERABILITY_CHECK_NAME]: {
    title: s__('SecurityApprovals|Vulnerability-Check'),
    popoverText: s__(
      'SecurityApprovals|A merge request approval is required when a security report contains a new vulnerability of high, critical, or unknown severity.',
    ),
    documentationText: s__('SecurityApprovals|Learn more about Vulnerability-Check'),
  },
  [LICENSE_CHECK_NAME]: {
    title: s__('SecurityApprovals|License-Check'),
    popoverText: s__(
      'SecurityApprovals|A merge request approval is required when the license compliance report contains a denied license.',
    ),
    documentationText: s__('SecurityApprovals|Learn more about License-Check'),
  },
  [COVERAGE_CHECK_NAME]: {
    title: s__('SecurityApprovals|Coverage-Check'),
    popoverText: s__(
      'SecurityApprovals|A merge request approval is required when test coverage declines.',
    ),
    documentationText: s__('SecurityApprovals|Learn more about Coverage-Check'),
  },
};

export const APPROVALS_HELP_PATH = 'user/project/merge_requests/merge_request_approvals';

export const APPROVAL_SETTINGS_I18N = {
  authorApprovalLabel: s__('ApprovalSettings|Prevent MR approvals by the author.'),
  preventMrApprovalRuleEditLabel: s__(
    'ApprovalSettings|Prevent users from modifying MR approval rules.',
  ),
  preventCommittersApprovalLabel: s__(
    'ApprovalSettings|Prevent approval of merge requests by merge request committers.',
  ),
  requireUserPasswordLabel: s__('ApprovalSettings|Require user password for approvals.'),
  removeApprovalsOnPushLabel: s__(
    'ApprovalSettings|Remove all approvals in a merge request when new commits are pushed to its source branch.',
  ),
  saveChanges: __('Save changes'),
  loadingErrorMessage: s__(
    'ApprovalSettings|There was an error loading merge request approval settings.',
  ),
  savingErrorMessage: s__(
    'ApprovalSettings|There was an error updating merge request approval settings.',
  ),
  savingSuccessMessage: s__('ApprovalSettings|Merge request approval settings have been updated.'),
};
