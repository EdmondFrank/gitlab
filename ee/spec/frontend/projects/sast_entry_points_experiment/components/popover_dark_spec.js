import '~/commons';
import { GlPopover, GlButton } from '@gitlab/ui';
import Cookies from 'js-cookie';
import PopoverDark from 'ee/projects/sast_entry_points_experiment/components/popover_dark.vue';
import { COOKIE_NAME } from 'ee/projects/sast_entry_points_experiment/constants';
import { mountExtended } from 'helpers/vue_test_utils_helper';
import ExperimentTracking from '~/experimentation/experiment_tracking';

jest.mock('~/experimentation/experiment_tracking');

let wrapper;

const sastDocumentationPath = 'sast_documentation_path';
const findPopover = () => wrapper.findComponent(GlPopover);
const findCtaButton = () => findPopover().findComponent(GlButton);
const findCloseButton = () => wrapper.findByTestId('close-btn');

function createComponent() {
  wrapper = mountExtended(PopoverDark, {
    propsData: { sastDocumentationPath },
  });
}

afterEach(() => {
  wrapper.destroy();
  Cookies.remove(COOKIE_NAME);
});

describe('When the cookie is set', () => {
  beforeEach(() => {
    Cookies.set(COOKIE_NAME, 'true');
    createComponent();
  });

  it('does not render the component', () => {
    expect(findPopover().exists()).toBe(false);
  });
});

describe('When the cookie is not set', () => {
  beforeEach(() => {
    createComponent();
  });

  it('renders the component', () => {
    expect(findPopover().exists()).toBe(true);
  });

  it('tracks the show event', () => {
    expect(ExperimentTracking.prototype.event).toHaveBeenCalledWith('show');
  });

  it('uses the sastDocumentationPath from the props for the button link', () => {
    expect(findCtaButton().attributes('href')).toBe(sastDocumentationPath);
  });

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('When clicking the CTA button', () => {
    beforeEach(() => {
      findCtaButton().vm.$emit('click');
    });

    it('tracks the cta_clicked event', () => {
      expect(ExperimentTracking.prototype.event).toHaveBeenCalledWith('cta_clicked');
    });
  });

  describe('When dismissing the component', () => {
    beforeEach(() => {
      findCloseButton().vm.$emit('click');
    });

    it('tracks the dismissed event', () => {
      expect(ExperimentTracking.prototype.event).toHaveBeenCalledWith('dismissed');
    });

    it('sets a cookie', () => {
      expect(Cookies.get(COOKIE_NAME)).toBe('true');
    });

    it('hides the component', () => {
      expect(findPopover().exists()).toBe(false);
    });
  });
});
