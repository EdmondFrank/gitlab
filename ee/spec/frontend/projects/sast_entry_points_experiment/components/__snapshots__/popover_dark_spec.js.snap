// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the cookie is not set matches the snapshot 1`] = `
<div
  class="gl-popover"
  offset="93"
  show=""
>
  <div
    class="gl-display-flex gl-mt-n2"
  >
    <img
      class="gl-ml-2 gl-mr-3"
      height="24"
      width="24"
    />
     
    <div>
      <div
        class="gl-font-weight-bold gl-font-lg gl-line-height-20 gl-text-theme-indigo-900 gl-mb-3"
      >
        
        Catch your security vulnerabilities ahead of time!
      
      </div>
       
      <div
        class="gl-font-base gl-line-height-20 gl-mb-3"
      >
        
        GitLab can scan your code for security vulnerabilities. Static Application Security Testing (SAST) helps you worry less and build more.
      
      </div>
       
      <a
        class="btn btn-link btn-md gl-button"
        href="sast_documentation_path"
      >
        <!---->
         
        <!---->
          
        <span
          class="gl-button-text"
        >
          
        How do I set up SAST?
      
        </span>
      </a>
    </div>
     
    <button
      aria-label="Close"
      class="btn gl-align-self-start gl-mt-n3 gl-mr-n3 btn-default btn-md gl-button btn-default-tertiary btn-icon"
      data-testid="close-btn"
      type="button"
    >
      <!---->
       
      <svg
        aria-hidden="true"
        class="gl-button-icon gl-icon s16"
        data-testid="close-icon"
        role="img"
      >
        <use
          href="#close"
        />
      </svg>
        
      <!---->
    </button>
  </div>
</div>
`;
